File: topics/program-creation/examples/shape_drawing.c
/*
* Program: shape_drawing.c
* Draws a number of shapes to the screen using SwinGame.
*/
#include <stdio.h>
#include <stdbool.h>
#include "SwinGame.h"

int main()
{
    open_graphics_window("Shape Drawing", 800, 600);
    load_default_colors();
    
    clear_screen();
    
    fill_rectangle(ColorWhite, 10, 10, 780, 580);
    
    refresh_screen();
    delay(500);
    
    fill_circle(ColorRed, 50, 50, 25);
    fill_circle(ColorGreen, 80, 50, 25);
    fill_circle(ColorBlue, 110, 50, 25);
    
    refresh_screen();
    delay(500);
    
    fill_triangle(ColorYellow, 100, 100, 150, 175, 210, 115);
    refresh_screen();
    delay(2000);
    
    release_all_resources();
    return 0;
}

File: topics/program-creation/examples/ShapeDrawing.pas
//
// Program: ShapeDrawing.pas
// Draws a number of shapes to the screen using SwinGame.
//
program GameMain;
uses
  sgTypes, sgInput, sgAudio, sgGraphics, sgResources, sgUtils, sgText;

begin
    OpenGraphicsWindow('Shape Drawing', 800, 600);

    ClearScreen();

    FillRectangle(ColorWhite, 10, 10, 780, 580);

    RefreshScreen();
    Delay(500);

    FillCircle(ColorRed, 50, 50, 25);
    FillCircle(ColorGreen, 80, 50, 25);
    FillCircle(ColorBlue, 110, 50, 25);

    RefreshScreen();
    Delay(500);

    FillTriangle(ColorYellow, 100, 100, 150, 175, 210, 115);
    RefreshScreen();
    Delay(2000);

    ReleaseAllResources();
end.
